guava/src/com/google/common/math/BigIntegerMath.java

```diff
@@ -332,7 +337,10 @@ public static BigInteger divide(BigInteger p, BigInteger q, RoundingMode mode) {
    *
    * @throws IllegalArgumentException if {@code n < 0}
    */
-  public static BigInteger factorial(int n) {
+  @SuppressWarnings(value = {"lowerbound:assignment.type.incompatible",// line 350: Since `n` is required to be non negative
+          // IntMath.divide() will also return a non negative value
```

Comment:
Rather than just asserting this fact, you should explain why it is true in this case but not in general. So here that would be something like "IntMath.divide return non-negative if both the divisor (n * IntMath.log2(n, CEILING)) and the dividend (Long.SIZE? or CEILING, whichever is actually correct) are non-negative."